tardis/visualization/tools/sdec_plot.py

```diff
@@ -991,46 +1089,53 @@ def generate_plot_ply(
-            value is 'virtual'
-        packet_wvl_range : astropy.Quantity or None, optional
-            Wavelength range to restrict the analysis of escaped packets. It
-            should be a quantity having units of Angstrom, containing two
-            values - lower lambda and upper lambda i.e.
-            [lower_lambda, upper_lambda] * u.AA. Default value is None
-        distance : astropy.Quantity or None, optional
-            Distance used to calculate flux instead of luminosity in the plot.
-            It should have a length unit like m, Mpc, etc. Default value is None
-        show_modeled_spectrum : bool, optional
-            Whether to show modeled spectrum in SDEC Plot. Default value is
-            True
-        fig : plotly.graph_objs._figure.Figure or None, optional
-            Figure object on which to create plot. Default value is None which
-            will create plot on a new Figure object.
-        graph_height : int, optional
-            Height (in px) of the plotly graph to display. Default value is 600
-        cmapname : str, optional
-            Name of the colormap to be used for showing elements.
-            Default value is "jet"
-
-        Returns
-        -------
-        plotly.graph_objs._figure.Figure
-            Figure object on which SDEC Plot is created
+         Generate interactive Spectral element DEComposition (SDEC) Plot using plotly.
+
+         Parameters
+         ----------
+         packets_mode : {'virtual', 'real'}, optional
+             Mode of packets to be considered, either real or virtual. Default
+             value is 'virtual'
+         packet_wvl_range : astropy.Quantity or None, optional
+             Wavelength range to restrict the analysis of escaped packets. It
+             should be a quantity having units of Angstrom, containing two
+             values - lower lambda and upper lambda i.e.
+             [lower_lambda, upper_lambda] * u.AA. Default value is None
+         distance : astropy.Quantity or None, optional
+             Distance used to calculate flux instead of luminosity in the plot.
+             It should have a length unit like m, Mpc, etc. Default value is None
+         show_modeled_spectrum : bool, optional
+             Whether to show modeled spectrum in SDEC Plot. Default value is
+             True
+         fig : plotly.graph_objs._figure.Figure or None, optional
+             Figure object on which to create plot. Default value is None which
+             will create plot on a new Figure object.
+         graph_height : int, optional
+             Height (in px) of the plotly graph to display. Default value is 600
+         cmapname : str, optional
+             Name of the colormap to be used for showing elements.
+             Default value is "jet"
+        nelements: int
+            Number of elements to include in plot. Determined by the
+            contribution to total lumionsity absorbed and emitted.
```

Comment:
*luminosities

Reply:
Luminosity is correct because it's the total summed across all wavelength bins, absorption, and emission.